sgpt/cache.py

```diff
@@ -83,16 +85,36 @@ def __call__(self, func: Callable) -> Callable:
         :return: Wrapped function with chat caching.
         """
         def wrapper(*args, **kwargs):
+            refine = kwargs.pop("refine", False)
             chat_id = kwargs.pop("chat_id", None)
-            message = {"role": "user", "content": kwargs.pop("message")}
+
+            # Decide which cache file to read from, and which to write to
+            read_chat_id = chat_id
+            write_chat_id = chat_id
             if not chat_id:
-                kwargs["message"] = [message]
-                return func(*args, **kwargs)
-            kwargs["message"] = self._read(chat_id)
+                if not refine:
+                    # Starting a new unnamed chat
+                    read_chat_id = None
+                    write_chat_id = "unnamed"
+                else:
+                    chat_id = self._read_current_chatid()
+                    if chat_id == "unnamed":
+                        # First reply to the unnamed chat
+                        read_chat_id = "unnamed"
+                        write_chat_id = "".join(choice(hexdigits) for _ in range(32))
+                        #print("Converting " + read_chat_id + " chat into: " + write_chat_id)
```

Comment:
This will rename the `unnamed` chat to a random id, the first time an `unnamed` chat is refined.

I would not mind removing this, to simplify things. I'm not sure how beneficial it is.

We could just keep refinements in the `unnamed` chat session until a new session is started.